### PR TITLE
Adding Consent and Authentication Webapp Implementation

### PR DESCRIPTION
**Pull request by Ashi1993**
## Adding Consent and Authentication Webapp Implementation

> This PR adds Consent and Authentication Webapp Implementation

**Issue link:** *https://github.com/wso2-enterprise/solutions-bfsi/issues/20*

**Doc Issue:** *Optional, link issue from [documentation repository](https://github.com/wso2/docs-ob/issues)*

**Applicable Labels:** *Spec, product, version, type (specify requested labels)*

------

### Development Checklist

1. [x] Build complete solution with pull request in place.
2. [x] Ran checkstyle plugin with pull request in place.
3. [x] Ran Findbugs plugin with pull request in place.
4. [x] Ran FindSecurityBugs plugin and verified report.
5. [x] Formatted code according to WSO2 code style.
6. [x] Have you verified the PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?
7. [ ] Migration scripts written (if applicable).
8. [x] Have you followed secure coding standards in [WSO2 Secure Engineering Guidelines](http://wso2.com/technical-reports/wso2-secure-engineering-guidelines)?

### Testing Checklist

1. [ ] Written unit tests.
2. [ ] Verified tests in multiple database environments (if applicable).
3. [ ] Tested with BI enabled  (if applicable).

### CODE REVIEW

File: financial-services-accelerator/components/org.wso2.financial.services.accelerator.common/src/main/java/org/wso2/financial/services/accelerator/common/config/FinancialServicesConfigParser.java [hunk 1]
```diff
@@ -160,25 +161,26 @@ private void buildFSExecutors() {
...
+                // obtaining each consent type element under FinancialServicesGatewayExecutors
+                // tag
                 Iterator consentTypeElement = financialServicesGatewayExecutors.getChildElements();
                 while (consentTypeElement.hasNext()) {
                     OMElement consentType = (OMElement) consentTypeElement.next();
                     String consentTypeName = consentType.getLocalName();
                     Map<Integer, String> executors = new HashMap<>();
-                    //obtaining each Executor element under each consent type
+                    // obtaining each Executor element under each consent type
                     Iterator<OMElement> obExecutor = consentType.getChildrenWithName(
```
Comment (aka4rKO) [inline financial-services-accelerator/components/org.wso2.financial.services.accelerator.common/src/main/java/org/wso2/financial/services/accelerator/common/config/FinancialServicesConfigParser.java:172]: should we rename this to something other than ob?

File: financial-services-accelerator/components/org.wso2.financial.services.accelerator.common/src/test/java/org/wso2/financial/services/accelerator/common/test/FSConfigParserTests.java [hunk 2]
```diff
@@ -68,16 +69,16 @@ public void testConfigParserInit() {
         Assert.assertEquals(openBankingConfigParser.getConfiguration().get("Sample.FSHandler"), "DummyValue");
         Assert.assertEquals(openBankingConfigParser.getConfiguration().get("Sample.FSHandler2"), "property.value");
         Assert.assertNotNull(openBankingConfigParser.getConfiguration().get("Sample.FSHandler4"));
-        Map<String, Map<Integer, String>> openBankingExecutors =
-                FinancialServicesConfigParser.getInstance().getFinancialServicesExecutors();
+        Map<String, Map<Integer, String>> openBankingExecutors = FinancialServicesConfigParser.getInstance()
```
Comment (aka4rKO) [inline financial-services-accelerator/components/org.wso2.financial.services.accelerator.common/src/test/java/org/wso2/financial/services/accelerator/common/test/FSConfigParserTests.java:72]: here as well

File: financial-services-accelerator/internal-apis/org.wso2.financial.services.accelerator.authentication.endpoint/src/main/webapp/cookie_policy.jsp [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+          <div class="col-xs-12 col-sm-12 col-md-9 col-lg-9">
+              <div>
+                  <h1 class="wr-title gray-bg white">
+                      <%=i18n(resourceBundle, "wso2.open.banking")%> - <%=i18n(resourceBundle, "privacy.policy.cookies")%>
+                  </h1>
+              </div>
+              <!-- Customizable content. Due to this nature, i18n is not implemented for this section -->
+              <div id="cookiePolicy" class="policies-wrapper">
+                  <section>
+                      <h4><a href="https://wso2.com/solutions/financial/open-banking/"><strong>About WSO2 Open Banking Solution</strong></a></h4>
```
Comment (aka4rKO) [inline financial-services-accelerator/internal-apis/org.wso2.financial.services.accelerator.authentication.endpoint/src/main/webapp/cookie_policy.jsp:55]: are these ok to have?

Comment (aka4rKO) [reply financial-services-accelerator/internal-apis/org.wso2.financial.services.accelerator.authentication.endpoint/src/main/webapp/cookie_policy.jsp:55]: check other places in this file

File: financial-services-accelerator/internal-apis/org.wso2.financial.services.accelerator.authentication.endpoint/src/main/webapp/images/logo-dark.svg
Comment (aka4rKO) [inline financial-services-accelerator/internal-apis/org.wso2.financial.services.accelerator.authentication.endpoint/src/main/webapp/images/logo-dark.svg:1]: is this image ok to have in this repo?

File: financial-services-accelerator/internal-apis/org.wso2.financial.services.accelerator.authentication.endpoint/src/main/webapp/images/wso2-open-banking-logo.png
Comment (aka4rKO) [inline financial-services-accelerator/internal-apis/org.wso2.financial.services.accelerator.authentication.endpoint/src/main/webapp/images/wso2-open-banking-logo.png:1]: this image also has open banking

File: financial-services-accelerator/internal-apis/org.wso2.financial.services.accelerator.authentication.endpoint/src/main/webapp/includes/footer.jsp [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ ~ KIND, either express or implied. See the License for the
+ ~ specific language governing permissions and limitations
+ ~ under the License.
+ --%>
+
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<footer class="footer">
+    <div class="container-fluid">
+        <p>WSO2 Open Banking <script>document.write(new Date().getFullYear());</script>
```
Comment (aka4rKO) [inline financial-services-accelerator/internal-apis/org.wso2.financial.services.accelerator.authentication.endpoint/src/main/webapp/includes/footer.jsp:22]: here as well

File: financial-services-accelerator/internal-apis/org.wso2.financial.services.accelerator.authentication.endpoint/src/main/webapp/cookie_policy.jsp (hunk 3 above)
Comment (Ashi1993) [reply financial-services-accelerator/internal-apis/org.wso2.financial.services.accelerator.authentication.endpoint/src/main/webapp/cookie_policy.jsp:55]: yes still we have not decided to change these

File: financial-services-accelerator/internal-apis/org.wso2.financial.services.accelerator.authentication.endpoint/src/main/webapp/images/logo-dark.svg
Comment (Ashi1993) [reply financial-services-accelerator/internal-apis/org.wso2.financial.services.accelerator.authentication.endpoint/src/main/webapp/images/logo-dark.svg:1]: Still we have not decided to change the naming open banking from the webapps

File: financial-services-accelerator/internal-apis/org.wso2.financial.services.accelerator.authentication.endpoint/src/main/webapp/images/wso2-open-banking-logo.png
Comment (Ashi1993) [reply financial-services-accelerator/internal-apis/org.wso2.financial.services.accelerator.authentication.endpoint/src/main/webapp/images/wso2-open-banking-logo.png:1]: Still we have not decided to change the naming open banking from the webapps

File: financial-services-accelerator/internal-apis/org.wso2.financial.services.accelerator.authentication.endpoint/src/main/webapp/includes/footer.jsp (hunk 4 above)
Comment (Ashi1993) [reply financial-services-accelerator/internal-apis/org.wso2.financial.services.accelerator.authentication.endpoint/src/main/webapp/includes/footer.jsp:22]: Still we have not decided to change the naming open banking from the webapps

File: financial-services-accelerator/internal-apis/org.wso2.financial.services.accelerator.authentication.endpoint/src/main/webapp/libs/bootstrap_3.4.1/js/npm.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+require('../../js/button.js')
+require('../../js/carousel.js')
+require('../../js/collapse.js')
+require('../../js/dropdown.js')
+require('../../js/modal.js')
+require('../../js/tooltip.js')
+require('../../js/popover.js')
+require('../../js/scrollspy.js')
+require('../../js/tab.js')
+require('../../js/affix.js')
```
Comment (aka4rKO) [inline financial-services-accelerator/internal-apis/org.wso2.financial.services.accelerator.authentication.endpoint/src/main/webapp/libs/bootstrap_3.4.1/js/npm.js:13]: lets add an EOF line here

File: financial-services-accelerator/components/org.wso2.financial.services.accelerator.common/src/test/java/org/wso2/financial/services/accelerator/common/test/FSConfigParserTests.java (hunk 2 above)
Comment (Ashi1993) [reply financial-services-accelerator/components/org.wso2.financial.services.accelerator.common/src/test/java/org/wso2/financial/services/accelerator/common/test/FSConfigParserTests.java]: FIxed in https://github.com/wso2/financial-services-accelerator/pull/110/commits/fe1e2b6aa311de998ac4f0028ad72f9247c4dd0c

File: financial-services-accelerator/components/org.wso2.financial.services.accelerator.common/src/main/java/org/wso2/financial/services/accelerator/common/config/FinancialServicesConfigParser.java (hunk 1 above)
Comment (Ashi1993) [reply financial-services-accelerator/components/org.wso2.financial.services.accelerator.common/src/main/java/org/wso2/financial/services/accelerator/common/config/FinancialServicesConfigParser.java]: FIxed in https://github.com/wso2/financial-services-accelerator/pull/110/commits/fe1e2b6aa311de998ac4f0028ad72f9247c4dd0c

File: financial-services-accelerator/internal-apis/org.wso2.financial.services.accelerator.authentication.endpoint/src/main/webapp/libs/bootstrap_3.4.1/js/npm.js (hunk 5 above)
Comment (Ashi1993) [reply financial-services-accelerator/internal-apis/org.wso2.financial.services.accelerator.authentication.endpoint/src/main/webapp/libs/bootstrap_3.4.1/js/npm.js]: FIxed in https://github.com/wso2/financial-services-accelerator/pull/110/commits/33337e150774c188bd8783e602cf48d83f28ff56

**Review by aka4rKO — APPROVED** [review]

**Review by hasithakn — APPROVED** [review]

**Pull request merged**